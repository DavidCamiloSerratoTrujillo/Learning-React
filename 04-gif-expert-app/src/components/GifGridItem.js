import React from 'react';
import PropTypes from 'prop-types'
export const GifGridItem = ({title,url}) => {
  return (
    <div className='card cola cao'>
        <img src ={url} alt ={title}/>{/*Se cargan las imagenes*/}
        <p>{title}</p> 
    </div>
  )
}

GifGridItem.propTypes ={
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}