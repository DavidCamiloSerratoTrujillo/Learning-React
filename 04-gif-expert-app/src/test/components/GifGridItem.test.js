import{shallow} from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';
import React from 'react';

import '@testing-library/jest-dom'

describe("Probar GridGifItem",()=>{
    const title = "Soy un titulo";
    const url = 'https://localhost/algo.jpg;'
    let wrapper= shallow (<GifGridItem title = {title} url = {url}/>);
    beforeEach(()=>{
        wrapper = shallow (<GifGridItem title = {title} url = {url}/>);

    });
    test('should show the component correctly', () => { 

    expect(wrapper).toMatchSnapshot();

     });
     test('Debe tener un parrafo con el title',()=>{
        const textoParrafo = wrapper.find('p').text().trim()
        expect(textoParrafo).toBe(title);
     });
     test('debe de tener la imagen igual al url y alt de los props', () => { 
        const img = wrapper.find('img');
        // const img = wrapper.find('img').props().src;
        // const img = wrapper.find('img').prop('src');
        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
      })
      test('Devb tener card', () => { 
       const div = wrapper.find('div')
       const classname = div.prop('className');
       expect(classname.includes('card')).not.toBe(false);
      });

});