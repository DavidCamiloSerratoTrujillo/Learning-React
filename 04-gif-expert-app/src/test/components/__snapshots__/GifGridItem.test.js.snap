// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar GridGifItem should show the component correctly 1`] = `
<div
  className="card cola cao"
>
  <img
    alt="Soy un titulo"
    src="https://localhost/algo.jpg;"
  />
  <p>
    Soy un titulo
  </p>
</div>
`;
